pkg/pgmodel/ingestor/metadata-dispatcher.go

```diff
@@ -0,0 +1,77 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package ingestor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+const backoffDuration = time.Millisecond * 500
+
+type metadataDispatcher struct {
+	conn pgxconn.PgxConn
+}
+
+func newMetadataDispatcher(conn pgxconn.PgxConn) *metadataDispatcher {
+	return &metadataDispatcher{conn}
+}
+
+// CompleteMetricCreation to implement the Dispatcher interface.
+func (m *metadataDispatcher) CompleteMetricCreation() error { return nil }
+
+// Close to implement the Dispatcher interface.
+func (m *metadataDispatcher) Close() {}
+
+func (m *metadataDispatcher) Insert(data interface{}) (uint64, error) {
+	rows, ok := data.([]model.Metadata)
+	if !ok {
+		panic("invalid data received")
+	}
+	totalRows := uint64(len(rows))
+	// Ingest metadata.
+	insertedRows, err := insertMetadataWithBackoff(m.conn, rows)
+	if err != nil {
+		return insertedRows, err
+	}
+	if totalRows != insertedRows {
+		return insertedRows, fmt.Errorf("failed to insert all metadata: inserted %d rows out of %d rows in total", insertedRows, totalRows)
+	}
+	return insertedRows, nil
+}
+
+func insertMetadataWithBackoff(conn pgxconn.PgxConn, data []model.Metadata) (uint64, error) {
```

Comment:
This does not insert with backoff, it just sleeps for half of second before inserting. Also, does not retry like the comment below says.

Is there a reason for introducing retrying with backoff?

Reply:
> does not insert with backoff

I wrote it as backoff as what prometheus calls a backoff. 

> Is there a reason for introducing retrying with backoff

Network connections, interrupts, can be some